src/App.tsx

```diff
@@ -15,71 +16,107 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+interface States {
+  selectedGood: string;
+}
+
+export class App extends Component<{}, States> {
+  state = {
+    selectedGood: 'Jam',
+  };
+
+  select = (good: string) => {
+    if (good === this.state.selectedGood) {
+      this.setState({
+        selectedGood: '',
+      });
+
+      return;
+    }
+
+    this.setState({
+      selectedGood: good,
+    });
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+    let title = <h1 className="title">No goods selected</h1>;
+
+    if (selectedGood) {
+      title = (
+        <h1 className="title is-flex is-align-items-center">
+          {`${selectedGood} is selected`}
+
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            onClick={() => {
+              this.setState({ selectedGood: '' });
+            }}
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+          />
+        </h1>
+      );
+    }
+
+    return (
+      <main className="section container">
+
+        {title}
+
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              return (
+                <tr
+                  data-cy="Good"
+                  className={cd('', {
+                    'has-background-success-light': good === selectedGood,
```

Comment:
you checked is current good selected several times, create separate function for this purpose, don't repeat yourself